pkg/kubelet/eviction/eviction_manager.go

```diff
@@ -209,6 +216,14 @@ func (m *managerImpl) synchronize(diskInfoProvider DiskInfoProvider, podFunc Act
 	// record an event about the resources we are now attempting to reclaim via eviction
 	m.recorder.Eventf(m.nodeRef, api.EventTypeWarning, "EvictionThresholdMet", "Attempting to reclaim %s", resourceToReclaim)
 
+	// check if there are node-level resources we can reclaim to reduce pressure before evicting end-user pods.
+	if m.reclaimNodeLevelResources(resourceToReclaim, observations) {
```

Comment:
resourceToReclaim is the first entry in starvedResources. In the case when there are multiple starved resources, only one resource is reclaimed. Is this intended? Why not reclaiming all the starved resources?